### DISCUSSION TITLE
mysql第一次启动时使用ORM链接会报Retrieval of the RSA public key is not enabled for insecure connections

### DISCUSSION THREAD

**Discussion by pya789** [dotnetcore/FreeSql#1546] [General]
本机mysql服务第一次启动时 使用ORM链接会报：
【主库】Next recovery time: 2023/6/23 18:37:03 (Retrieval of the RSA public key is not enabled for insecure connections.)

链接字符串：
"Data Source=127.0.0.1;Port=3306;User ID=root;Password=xxxx; Initial Catalog=xxxx;Charset=utf8; SslMode=none;Min pool size=1";

使用Navicat或者命令行登录一次mysql后  再使用ORM链接一切正常  
就是说第一次启动mysql服务 必须使用命令或者Navicat登录一次之后ORM才能正常使用 否则就报标题的错误。

**Comment by 2881099** [comment]
new MySqlConnection(...).Open()

这个测试通过，就行

**Comment by gitlsl** [comment]
加上  AllowPublicKeyRetrieval=true  这个参数就可以了

**Reply by pya789** [reply]
好的 谢谢
